src/models/cost_functions/ImportExportCost.jl

```diff
@@ -0,0 +1,259 @@
+    ImportExportCost()
+end
+
+"""Get [`ImportExportCost`](@ref) `import_offer_curves`."""
+get_import_offer_curves(value::ImportExportCost) = value.import_offer_curves
+"""Get [`ImportExportCost`](@ref) `export_offer_curves`."""
+get_export_offer_curves(value::ImportExportCost) = value.export_offer_curves
+"""Get [`ImportExportCost`](@ref) `ancillary_service_offers`."""
+get_ancillary_service_offers(value::ImportExportCost) = value.ancillary_service_offers
+"""Get [`ImportExportCost`](@ref) `energy_import_weekly_limit`."""
+get_energy_import_weekly_limit(value::ImportExportCost) = value.energy_import_weekly_limit
+"""Get [`ImportExportCost`](@ref) `energy_export_weekly_limits`."""
+get_energy_export_weekly_limit(value::ImportExportCost) = value.energy_export_weekly_limit
+
+"""Set [`ImportExportCost`](@ref) `import_offer_curves`."""
+set_import_offer_curves!(value::ImportExportCost, val) =
+    value.import_offer_curves = val
+"""Set [`ImportExportCost`](@ref) `export_offer_curves`."""
+set_export_offer_curves!(value::ImportExportCost, val) =
+    value.export_offer_curves = val
+"""Set [`ImportExportCost`](@ref) `ancillary_service_offers`."""
+set_ancillary_service_offers!(value::ImportExportCost, val) =
+    value.ancillary_service_offers = val
+"""Get [`ImportExportCost`](@ref) `energy_import_weekly_limit`."""
+set_energy_import_weekly_limit!(value::ImportExportCost, val) =
+    value.energy_import_weekly_limit = val
+"""Get [`ImportExportCost`](@ref) `energy_export_weekly_limits`."""
+set_energy_export_weekly_limit!(value::ImportExportCost, val) =
+    value.energy_export_weekly_limit = val
+
+function is_import_export_curve(curve::ProductionVariableCostCurve)
+    return (curve isa CostCurve{PiecewiseIncrementalCurve}) &&
+           iszero(get_initial_input(get_value_curve(curve)))
+end
+
+"""
+Make a CostCurve{PiecewiseIncrementalCurve} suitable for inclusion in a ImportExportCost from
+the FunctionData that might be used to store such a cost curve in a time series.
+"""
+function make_import_export_curve(
+    curve::PiecewiseStepData,
+    power_units::UnitSystem = UnitSystem.NATURAL_UNITS,
+)
+    cc = CostCurve(
+        PiecewiseIncrementalCurve(0.0, curve.x_coords, curve.y_coords),
+        power_units,
+    )
+    @assert is_import_export_curve(cc)
+    return cc
+end
+
+"""
+Make a CostCurve{PiecewiseIncrementalCurve} suitable for inclusion in a ImportExportCost from a
+vector of power values, a vector of costs, and an optional units system.
+
+# Examples
+```julia
+mbc = make_import_export_curve([0.0, 100.0, 105.0, 120.0, 130.0], [25.0, 26.0, 28.0, 30.0])
+mbc1 = make_import_export_curve([0.0, 100.0, 105.0, 120.0, 130.0], [25.0, 26.0, 28.0, 30.0], 10.0; power_units = UnitSystem.NATURAL_UNITS)
+```
+"""
+function make_import_export_curve(
+    powers::Vector{Float64},
+    prices::Vector{Float64},
+    power_units::UnitSystem = UnitSystem.NATURAL_UNITS,
+)
+    valid_data = (length(powers) == length(prices) + 1)
```

Comment:
Should probably validate that `iszero(first(powers))` since we're requiring that the curve start at the origin